dali/test/python/test_dali_tf_dataset.py

```diff
@@ -69,64 +81,102 @@ def define_graph(self):
             images, 
             crop_pos_x = self.uniform(),
             crop_pos_y = self.uniform())
-        im_ids_float = self.cast(im_ids)
+        if self.device is 'gpu':
+            im_ids = im_ids.gpu()
+        im_ids_16 = self.cast(im_ids)
 
         return (
             output, 
             im_ids,
-            im_ids_float)
+            im_ids_16)
+
+
+def _dataset_options():
```

Comment:
Is it used only for tests?

Reply:
Yes, it is needed to run this tests. Otherwise TF tries to apply some optimizations to the dataset and they are not available on the GPU.

Reply:
But the `dataset` will work without it outside the test?

Reply:
It will for the CPU. 
For now we don't run it anywhere else. When we will add some kind of example I will move it accordingly.